### PR TITLE
Moving all packages to pip & user application paths

### PR DESCRIPTION
**Pull request by eeintech**
* All the dependencies are installed from PyPI instead of relying on submodules
* There is a new settings menu for user configuration (Under "Settings > User"):

image

Those paths are telling Ki-nTree where all the user configuration files are stored, instead of being hard-coded.
By default the configuration files are created in the user's home folder.
The path to the configuration and cache files can be changed and they are stored in a `settings.yaml` file also located user's home folder (in `kintree` folder).

### CODE REVIEW

File: config/settings.py [hunk 1]
```diff
@@ -25,44 +37,44 @@ def enable_test_mode():
...
 sys.path.append(os.path.join(PROJECT_DIR, 'kicad'))
 # Tests
 sys.path.append(os.path.join(PROJECT_DIR, 'tests'))
 
+# HOME FOLDER
+USER_HOME = os.path.expanduser("~")
+# APP NAME
+APP_NAME = 'kintree'
+# CONFIG PATH
+HOME_DIR = os.path.join(USER_HOME, APP_NAME, '')
```
Comment (randrej) [inline config/settings.py:50]: I'd use the .config directory on Linux. Something like this should work:

    import platform  # this goes up top
    if platform.system() == 'Linux';
         HOME_DIR = os.path.join(USER_HOME, '.config', APP_NAME, '') 
    else:
        HOME_DIR = os.path.join(USER_HOME, APP_NAME, '')

**Review by randrej — COMMENTED** [review]
I suggest one small change to make this a bit more compatible with Linux conventions.

File: config/settings.py (hunk 1 above)
Comment (eeintech) [reply config/settings.py:50]: @randrej Is this convention good for Windows too? I imagine MacOS yes

Comment (eeintech) [reply config/settings.py:50]: Sorry I misread, this is only Linux so it should work

Comment (eeintech) [reply config/settings.py:50]: Rolling out this fix in #42 , thanks @randrej!

**Pull request merged**